loader/windows/icd_windows.c

```diff
@@ -103,18 +156,57 @@ BOOL CALLBACK khrIcdOsVendorsEnumerate(PINIT_ONCE InitOnce, PVOID Parameter, PVO
                 KHR_ICD_TRACE("Value not zero, skipping\n");
                 continue;
             }
-
             // add the library
-            khrIcdVendorAdd(cszLibraryName);
+            AdapterAdd(cszLibraryName, ZeroLuid);
         }
+    }
 
-        result = RegCloseKey(platformsKey);
-        if (ERROR_SUCCESS != result)
+    // Add adapters according to DXGI's preference order
+    HMODULE hDXGI = LoadLibrary("dxgi.dll");
+    if (hDXGI)
+    {
+        IDXGIFactory* pFactory = NULL;
+        PFN_CREATE_DXGI_FACTORY pCreateDXGIFactory = (PFN_CREATE_DXGI_FACTORY)GetProcAddress(hDXGI, "CreateDXGIFactory");
+        HRESULT hr = pCreateDXGIFactory(&IID_IDXGIFactory, &pFactory);
+        if (SUCCEEDED(hr))
         {
-            KHR_ICD_TRACE("Failed to close platforms key %s, ignoring\n", platformsName);
+            UINT i = 0;
+            IDXGIAdapter* pAdapter = NULL;
+            while (SUCCEEDED(pFactory->lpVtbl->EnumAdapters(pFactory, i++, &pAdapter)))
```

Comment:
Do we have to go through the table here and in line 178?
According to this you can call the method directly:
https://docs.microsoft.com/en-us/windows/win32/api/dxgi/nf-dxgi-idxgiadapter-getdesc
IDXGIAdapter * pDXGIAdapter;
pDXGIDevice->GetAdapter(&pDXGIAdapter);

without needing to pass "this".